h2o-core/src/main/java/water/rapids/ast/prims/reducers/AstAll.java

```diff
@@ -30,9 +32,14 @@ public int nargs() {
   public ValNum apply(Env env, Env.StackHelp stk, AstRoot asts[]) {
     Val val = stk.track(asts[1].exec(env));
     if (val.isNum()) return new ValNum(val.getNum() == 0 ? 0 : 1);
-    for (Vec vec : val.getFrame().vecs())
-      if (vec.nzCnt() + vec.naCnt() < vec.length())
+    for (Vec vec : val.getFrame().vecs()) {
+      String[] domainV = vec.domain();
+      if (!isBoolColumn(domainV))
+        return new ValNum(0);         // not a boolean column
+      long trueCount = domainV[0].equalsIgnoreCase("true")?(vec.length()-vec.nzCnt()):vec.nzCnt();
```

Comment:
shouldn't this be `(vec.length() - vec.nzCnt() - vec.naCnt())` ?

Reply:
When 0 is the true value, we want to reject anything that is non-zero.  We want naCnt since by definition, NA are okay.  Hence, we only need vec.length()-vec.nzCnt().